docs/reference/files/file-charmcraft-yaml.rst

```diff
@@ -0,0 +1,901 @@
+.. _file-charmcraft-yaml:
+.. highlight:: yaml
+
+``charmcraft.yaml``
+*******************
+
+..
+    AUTHOR NOTE:
+    The full list of keys is defined in the Charmcraft project (but this implies upstream keys from craft-application):
+    https://github.com/canonical/charmcraft/blob/3.2.0/charmcraft/models/project.py#L381-L1070
+
+    Technically, the only key required upfront is type. But then, depending on what you choose, other keys become required as well. (The required keys are the ones that are set to a value, unless that value is a pydantic.Field that doesn't have either a default or a default_factory parameter)
```

Comment:
We've been very consistent about 2 spaces for indentation. You may recall we recently discussed switching my YAML examples in Snapcraft back to 2 spaces.

Reply:
The current [configuration for markdown and rst](https://github.com/canonical/starbase/blob/c5c71673c8957fe8bd914f04b15e200403f58902/.editorconfig#L31-L32) in starbase is the global default of [`indent_size = 4`](https://github.com/canonical/starbase/blob/c5c71673c8957fe8bd914f04b15e200403f58902/.editorconfig#L8), so if we want to change that well need to change the standard and [the existing documentation files](https://github.com/canonical/charmcraft/blob/main/docs/reference/models/bases-charm.rst?plain=1), including [in Starbase itself](https://github.com/canonical/starbase/blob/c5c71673c8957fe8bd914f04b15e200403f58902/docs/release-notes/index.rst?plain=1#L55-L72).

Reply:
I've created [an issue on Starbase](https://github.com/canonical/starbase/issues/307) so we can get clarification for everything that needs changing here.